### DISCUSSION TITLE
AttributeError: module 'skdatasets.repositories' has no attribute 'cran'" }

### DISCUSSION THREAD

**Discussion by yn-zpc** [GAA-UAM/scikit-fda#640] [Q&A]
Hello, I encountered an issue while running an example from the scikit-fda documentation. Specifically, when I try to load the phoneme dataset fetch_phoneme, I get an AttributeError.
`import skfda

# Trying to load the phoneme dataset
X, y = skfda.datasets.fetch_phoneme(return_X_y=True)`

AttributeError                            Traceback (most recent call last)
Cell In[2], line 3
      1 max_basis = 9
----> 3 X, y = skfda.datasets.fetch_phoneme(return_X_y=True)
      5 # Select only the first 5 samples
      6 X = X[:5]

File c:\Users\zhoupuchen\AppData\Local\Programs\Python\Python312\Lib\site-packages\skfda\datasets\_real_datasets.py:341, in fetch_phoneme(return_X_y, as_frame)
    332 """
    333 Load the phoneme dataset.
    334 
   (...)
    337 
    338 """
    339 descr = _phoneme_descr
--> 341 raw_dataset = _fetch_elem_stat_learn("phoneme")
    343 data = raw_dataset["phoneme"]
    345 n_points = 256

File c:\Users\zhoupuchen\AppData\Local\Programs\Python\Python312\Lib\site-packages\skfda\datasets\_real_datasets.py:229, in _fetch_elem_stat_learn(name)
    228 def _fetch_elem_stat_learn(name: str) -> Any:
--> 229     return _fetch_cran_no_encoding_warning(
    230         name,
    231         "ElemStatLearn",
...
    131     converter=converter,
    132     **kwargs,
    133 )

AttributeError: module 'skdatasets.repositories' has no attribute 'cran'

I installed the library using pip install scikit-fda.
I suspect there might be an issue with the skdatasets dependency.
I am unsure if there is a workaround or if my installation is missing something.

Is this a known issue with skfda or its dependencies?
Should I downgrade my Python version or a specific package to avoid this error?
Are there any alternative ways to access the fetch_phoneme dataset?
Thank you in advance for your help!

**Comment by vnmabus** [comment]
Could you please post your version of scikit-fda and scikit-datasets?

**Reply by yn-zpc** [reply]
The scikit-fda version is 0.9.1. The scikit-datasets version is 0.2.4

**Reply by vnmabus** [reply]
Maybe it is better to know also the version of the `numpy`, `pandas`, `scikit-learn` and `rdata` packages. When installing `scikit-fda` it should have installed `scikit-datasets[cran]`, which in turn should have installed these.

The module `cran` is only present if there is no error importing some functionally from these packages, so it can either happen that some of these got removed/downgraded or that a new version removed some functionality.

**Reply by vnmabus** [reply]
Ok, no, disregard the above. It seems that the culprit is indeed `distutils`, which is no longer available in Python 3.12. We need to patch `scikit-datasets` to remove this dependency.

**Reply by vnmabus** [reply]
Please see https://stackoverflow.com/a/77284076/2455333 for possible workarounds in the meantime.

**Reply by yn-zpc** [reply]
okkk, thank you for your help
